smolyay/samples.py

```diff
@@ -508,3 +508,620 @@ def _create(self):
                 2 * numpy.pi * indexes / num_points
             )
         self._points = self._scale_to_domain(points, [0, 2 * numpy.pi])
+
+
+import abc
+import itertools
+
+import numpy
+import scipy.stats.qmc
+
+
+class MultidimensionalPointSet(abc.ABC):
+    """Multidimensional set of points
+
+    Creates a set of grid points within the specified domain
+
+    Parameters
+    ----------
+    domain: list
+         the domain of the grid
+    """
+
+    def __init__(self):
+        self._domain = None
+        self._points = None
+        self._valid_cache = False
+
+    @property
+    def num_dimensions(self):
+        """int: number of independent variables."""
+        return numpy.shape(self.domain)[0]
```

Comment:
`domain` should be a numpy array, so this will work
```suggestion
        return self.domain.shape[0]
```